### PR TITLE
Fetch recommendations from DB and save to local storage

### PR DESCRIPTION
**Pull request by bharathappali**

### CODE REVIEW

File: src/main/java/com/autotune/database/service/ExperimentDBService.java [hunk 1]
```diff
@@ -135,6 +136,24 @@ public void loadAllExperimentsData() throws Exception {
             }
         }
         experimentInterface.addResultsToLocalStorage(KruizeOperator.autotuneObjectMap, resultDataList);
+
+        // Load Recommendations from DB and save to local
+        List<KruizeRecommendationEntry> recommendationEntries = experimentDAO.loadAllRecommendations();
+        if (null != recommendationEntries && !recommendationEntries.isEmpty()) {
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/database/service/ExperimentDBService.java:142]: Can you please add this null check for above 2 block also

File: src/main/java/com/autotune/database/service/ExperimentDBService.java [hunk 2]
```diff
@@ -135,6 +136,24 @@ public void loadAllExperimentsData() throws Exception {
...
+            List<ListRecommendationsAPIObject> recommendationsAPIObjects
+                    = null;
+            try {
+                recommendationsAPIObjects = DBHelpers.Converters.KruizeObjectConverters
+                    .convertRecommendationEntryToRecommendationAPIObject(recommendationEntries);
+            } catch (InvalidConversionOfRecommendationEntryException e) {
+                e.printStackTrace();
+            }
+            if (null != recommendationsAPIObjects && !recommendationsAPIObjects.isEmpty()) {
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/database/service/ExperimentDBService.java:151]: same please check and add for above block also

File: src/main/java/com/autotune/analyzer/experiment/ExperimentInterfaceImpl.java [hunk 3]
```diff
@@ -122,4 +125,36 @@ public boolean addResultsToLocalStorage(Map<String, KruizeObject> mainKruizeExpe
         return true;
     }
 
+    @Override
+    public boolean addRecommendationsToLocalStorage(Map<String, KruizeObject> mainKruizeExperimentMap,
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/experiment/ExperimentInterfaceImpl.java:129]: Can you use this function while generating recommendation for first time . Currently it is getting stored without this function , can you use this function there as well.  addRecommendationsToLocalStorage should be called at two place

File: src/main/java/com/autotune/analyzer/exceptions/InvalidConversionOfRecommendationEntryException.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+
+package com.autotune.analyzer.exceptions;
+
+public class InvalidConversionOfRecommendationEntryException extends Throwable{
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/exceptions/InvalidConversionOfRecommendationEntryException.java:20]: add this exception for ExperimentEntry and resultentry .. may be in separate PR

Comment (bharathappali) [reply src/main/java/com/autotune/analyzer/exceptions/InvalidConversionOfRecommendationEntryException.java:20]: Sure will raise a separate PR

File: src/main/java/com/autotune/database/service/ExperimentDBService.java (hunk 1 above)
Comment (bharathappali) [reply src/main/java/com/autotune/database/service/ExperimentDBService.java:153]: Added the null checks to the above blocks

File: src/main/java/com/autotune/database/service/ExperimentDBService.java (hunk 2 above)
Comment (bharathappali) [reply src/main/java/com/autotune/database/service/ExperimentDBService.java:162]: I have added the null checks for above blocks

**Review by dinogun — APPROVED** [review]
LGTM
